### ISSUE TITLE
Unable to use setValue for List

### ISSUE THREAD

**Issue by rajatvig** [realm/realm-cocoa#4335]
## Goals

Able to use `setValue` for List Types

## Expected Results

It should work

## Actual Results

Get Object Type errors

## Steps to Reproduce

run the sample code

## Code Sample

```swift
class BaseEntity: Object {
    dynamic var simpleProp: String?
    var listProp = List<ListEntity>()
}

class ListEntity: Object {
    dynamic var simpleProp: String?
}

let json: [String: Any] = ["simpleProp": "Wayne", "listProp": [["simpleProp": "Bruce"]]]

// this works
let entity1 = BaseEntity(value: json)
print(entity1)

// this does not
let entity = BaseEntity()

json.keys.forEach { key in
    entity.setValue(json[key], forKey: key)
}

print(entity)
```

## Version of Realm and Tooling

```
ProductName:	Mac OS X
ProductVersion:	10.12.2
BuildVersion:	16C41b

/Applications/Xcode.app/Contents/Developer
Xcode 8.1
Build version 8B62

/usr/local/var/rbenv/shims/pod
1.1.1
Realm (2.0.3)
RealmSwift (2.0.3)

/usr/local/bin/bash
GNU bash, version 4.4.0(1)-release (x86_64-apple-darwin16.0.0)

/usr/local/bin/carthage
0.18.1
(not in use here)

git: aliased to hub
git version 2.10.2
hub version 2.2.9
```

**Comment by austinzheng** [comment]
Thanks for reporting this issue. Tentatively, it looks like a bug:

In `RLMArray.mm`, the following is crashing:

```
static void RLMValidateMatchingObjectType(RLMArray *array, RLMObject *object) {
    if (!object) {
        @throw RLMException(@"Object must not be nil");
    }
    if (!object->_objectSchema) {
        @throw RLMException(@"Object cannot be inserted unless the schema is initialized. "
                            "This can happen if you try to insert objects into a RLMArray / List from a default value or from an overriden unmanaged initializer (`init()`).");
    }
    // crash here
    if (![array->_objectClassName isEqualToString:object->_objectSchema.className]) {
        @throw RLMException(@"Object type '%@' does not match RLMArray type '%@'.",
                            object->_objectSchema.className, array->_objectClassName);
    }
}
```

because `object->_objectSchema` is null.

**Comment by nabbestemmia** [comment]
Hello. I'm running into this issue as well on a simulator (iphone 5), and I just couldn't figure out how to resolve it. But then I tried to install my app on a different simulated device (iphone 5s), and it works fine. But if I try to install it again to the first siumulated device (the iphone 5) it still happens, even after I resetted its content and settings. Can someone help me to understand? Thank you
